gnat2goto/ireps/ireps_generator.py

```diff
@@ -1961,6 +2035,38 @@ def generate_code(self, optimize, schema_file_names):
         write(b, "end Follow_Irep;")
         write(b, "")
 
+        write_comment_block(b, "Remove_Bounds")
+        write(b, "function Remove_Bounds (I : Irep) return Irep")
+        write(b, "is")
+        write(b, "begin")
+        manual_indent(b)
+        write(b, "if I = 0 then")
+        with indent(b):
+            write(b, "return I;")
+        write(b, "end if;")
+        write(b, "")
+        write(b, "if Kind (I) = I_Bounded_Signedbv_Type then")
+        with indent(b):
+            write(b, "return Make_Signedbv_Type (Get_Subtype (I), Get_Width (I));")
+        write(b, "end if;")
+        write(b, "")
+        write(b, "declare")
+        with indent(b):
+            write(b, "N : Irep_Node renames Irep_Table.Table (I);")
+        write(b, "begin")
+        manual_indent(b)
```

Comment:
:pick: why use manual indent here?

Reply:
If I got the Python correctly, `with indent(b)` indents just the following line. `manual_indent(b)` indents until `manual_outdent(b)`.

Reply:
@xbauch `with_indent` will indent for the entire block, not just the next line (it's basically the python version of RAII); However, we might not want to have so many indented blocks so I'm definitely not insisting on changing this